pkg/operator/ceph/cluster/cluster.go

```diff
@@ -291,47 +282,22 @@ func clusterChanged(oldCluster, newCluster cephv1.ClusterSpec, clusterRef *clust
 }
 
 func (c *cluster) unsetOrchestrationRunning() {
-	c.orchRunMux.Lock()
+	c.orchMux.Lock()
 	c.orchestrationRunning = false
-	c.orchRunMux.Unlock()
-}
-
-func (c *cluster) setOrchestrationPending() {
-	c.orchPenMux.Lock()
-	c.orchestrationPending = true
-	c.orchPenMux.Unlock()
-}
-
-func (c *cluster) getOrchestrationRunning() bool {
-	c.orchRunMux.Lock()
-	defer c.orchRunMux.Unlock()
-	return c.orchestrationRunning
+	c.orchMux.Unlock()
 }
 
-func (c *cluster) getOrchestrationPending() bool {
-	c.orchPenMux.Lock()
-	defer c.orchPenMux.Unlock()
-	return c.orchestrationPending
-}
-
-// make checking and setting orchestrationRunning "atomic"
-func (c *cluster) checkSetOrchestrationRunning() bool {
-	defer c.orchRunMux.Unlock()
-	c.orchRunMux.Lock()
+func (c *cluster) checkSetOrchestrationStatus() bool {
+	defer c.orchMux.Unlock()
```

Comment:
This `defer` must be placed after the lock. If something were to go wrong and the function tried to unlock an already unlocked mutex, there is a panic: `fatal error: sync: unlock of unlocked mutex`

Reply:
Done